domain/secret/service/delete.go

```diff
@@ -35,7 +36,17 @@ func (s *SecretService) DeleteSecret(ctx context.Context, uri *secrets.URI, para
 	if err := s.canManage(ctx, uri, params.Accessor, params.LeaderToken); err != nil {
 		return errors.Trace(err)
 	}
+	return s.secretState.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
+		return s.InternalDeleteSecret(ctx, uri, params)
+	})
+}
 
+// InternalDeleteSecret removes the specified secret.
+// If revisions is nil or the last remaining revisions are removed.
+// It returns [secreterrors.PermissionDenied] if the secret cannot be managed by the accessor.
+// It is called by [DeleteSecret] on this service and also by the application service when deleting
```

Comment:
Should the application service call a method with owner arg(unit or app) instead of uri?

Reply:
It depends on your view on delineation of responsibilities. The application service will execute a query to load the uris of secrets owned by an app or unit. It's a simple bit of sql and it allows the DeleteSecret() on the secret service and the DeleteApplication() or DeleteUnit() on the application service to call a common method to delete a specified secret uri. 
Generally, deleting an application or unit will execute sql to delete records from a bunch of related tables for things like endpoints, port ranges, agent metadata etc. So there's already the requirement for the app service to have knowledge of how to deal with dependent records. For secrets, the delete is not being done but the records are being loaded to access the uris. The "easy" bits are all being done in the app service; the hard bits are being offloaded where necessary.

Reply:
We can do buck delete rather than fetch all URIs and then delete one URI by one.

Reply:
All of our service methods only take a single parameter to act on. That's the paradigm that is being used everywhere.
Even if a slice of uris was passed in, there'd still be individual calls to delete from the backend.